modules/context/repo.go

```diff
@@ -667,11 +667,38 @@ func RepoAssignment(ctx *Context) (cancel context.CancelFunc) {
 	}
 	ctx.Data["Tags"] = tags
 
-	brs, _, err := ctx.Repo.GitRepo.GetBranchNames(0, 0)
+	brs, err := git_model.FindBranchNames(ctx, git_model.FindBranchOptions{
+		RepoID:          ctx.Repo.Repository.ID,
+		IsDeletedBranch: util.OptionalBoolFalse,
+		ListOptions: db.ListOptions{
+			PageSize: -1,
+		},
+	})
 	if err != nil {
 		ctx.ServerError("GetBranches", err)
 		return
 	}
+	if len(brs) == 0 { // fallback to do a sync immediately
+		var doerID int64
+		if ctx.Doer != nil {
+			doerID = ctx.Doer.ID
+		}
+		if err := repo_module.SyncRepoBranches(ctx, ctx.Repo.Repository.ID, doerID); err != nil {
```

Comment:
It doesn't seem right. For example, if a read-only visitor user visits a repo, then the branch sync's doer is the non-related visitor user?